### PR TITLE
Fix #9324: crash trying to remove invalid footpath scenery.

### LINKED ISSUE #9324: BACKTRACE d91050 FootpathSceneryRemoveAction::Query
<table>
    <tr>
        <td></td>
        <td><a href='https://openrct2.sp.backtrace.io/dashboard/openrct2/project/openrct2/group/d91050213fecbd7e6421e2dfcffc537ee36f0d2f946dfd1051478762586c0f04?_t=W3sibmFtZSI6IndvcmtmbG93In1d'><b>New unique error: [openrct2]</b></a><br /><b>1</b> occurrence(s) <b></b> since (Tue May 28 2019 05:57:37)<br /></td>
    </tr>
</table>

<table>
<tr><td><strong>Application</strong></td><td>openrct2.exe</td></tr>

<tr><td><strong>Attributes</strong></td><td><strong>attachment_screenshot.png:</strong> C:\\Users\\barts\\Documents\\OpenRCT2\\screenshot\\Six Flags 2019-05-28 01-57-38.png: <em>1</em><br><strong>attachment_config.ini:</strong> C:\\Users\\barts\\Documents\\OpenRCT2\\e7b1eb9a-94b3-4623-8a30-6ad5a96578b5.ini: <em>1</em><br><strong>attachment_park.sv6.gz:</strong> C:\\Users\\barts\\Documents\\OpenRCT2\\e7b1eb9a-94b3-4623-8a30-6ad5a96578b5.sv6.gz: <em>1</em><br><strong>commit:</strong> 46889b5: <em>1</em><br><strong>attachment_park.sv6:</strong> *: <em>1</em><br></td></tr>
<tr><td><strong>Classifiers</strong></td><td><strong>invalid-read</strong> </td></tr>
<tr><td><strong>Frames</strong></td><td><code>FootpathSceneryRemoveAction::Query -> GameActions::QueryInternal -> GameActions::ExecuteInternal -> Network::ProcessGameCommands -> OpenRCT2::GameState::UpdateLogic -> OpenRCT2::GameState::Update -> OpenRCT2::Context::Update -> OpenRCT2::Context::RunGameLoop -> OpenRCT2::Context::RunOpenRCT2 -> NormalisedMain -> LaunchOpenRCT2 -> invoke_main</code></td></tr>
</table>

### PR DESCRIPTION
**Pull request by ZehMatt**
Just a missing nullptr check

### CODE REVIEW

File: src/openrct2/actions/FootpathSceneryRemoveAction.hpp [hunk 1]
```diff
@@ -69,8 +69,13 @@ DEFINE_GAME_ACTION(FootpathSceneryRemoveAction, GAME_COMMAND_REMOVE_FOOTPATH_SCE
         }
 
         auto tileElement = map_get_footpath_element(_loc.x / 32, _loc.y / 32, _loc.z / 8);
-        auto pathElement = tileElement->AsPath();
+        if (tileElement == nullptr)
+        {
+            log_error("Could not find path element.");
```
Comment (IntelOrca) [inline src/openrct2/actions/FootpathSceneryRemoveAction.hpp]: Should be `log_warning`, arguably do we need to log at all?

Comment (ZehMatt) [reply src/openrct2/actions/FootpathSceneryRemoveAction.hpp]: Not sure about the logging part but I'll change it to log_warning, should I also change the others? Also perhaps a thought from @duncanspumpkin ?

Comment (duncanspumpkin) [reply src/openrct2/actions/FootpathSceneryRemoveAction.hpp]: I prefer a log error for make results that don't have a good error message.

Comment (ZehMatt) [reply src/openrct2/actions/FootpathSceneryRemoveAction.hpp]: While you mention that, we should probably create a few issues where we feel like they deserve better error messages.

Comment (IntelOrca) [reply src/openrct2/actions/FootpathSceneryRemoveAction.hpp]: We are getting more and more error / warning lines showing up in the console, even just during the title screen parks. Any normal occurrence doesn't need to be logged.

Comment (duncanspumpkin) [reply src/openrct2/actions/FootpathSceneryRemoveAction.hpp]: They are basically dirty todo notes. I personally like that they are coming up more. It means we aren't doing things correctly and there are areas that need to be looked at.

**Comment by ZehMatt** [comment]
The discussion was never resolved and there is no final verdict, should we use log_error or log_warning?

**Comment by Gymnasiast** [comment]
To me, `log_warning` seems a comprise between `log_error` (too severe) and not reporting anything. So I would go for `log_warning`.

**Comment by ZehMatt** [comment]
Changed to it log_warning in Query

**Pull request merged**